### DISCUSSION TITLE
About build dds in linux with 4.11.0-14

### DISCUSSION THREAD

**Discussion by cloudandzack007** [OpenDDS/OpenDDS#3873] [Q&A] [answered]
Hello，I try to build opendds-3.21 at Linux(version is 4.11.0-14-generic)。I find many trouble，gcc&glic is too old。I am trying to fix this。But I want to know，if the opendds-3.21 can be build at this linux os。If not，which version can be used。my gcc version is 5.4.0，glibc version is 2.23

**Comment by mitza-oci** [comment, marked as answer]
Systems we test on are listed here https://opendds.org/documents/building.html but that doesn't mean it can't be ported to other platforms/versions.

Please send complete details of what you tried and the errors you saw.

**Reply by cloudandzack007** [reply]
I have select another version(dds 3.13.2)。It can be bulit at my os without update gcc&glibc。Thank you。

**Reply by jwillemsen** [reply]
That version is ancient, you should be able to use GCC 5.4 with newer versions of OpenDDS, maybe try the doc_group version of ACE/TAO, gcc 5.4 does have C++11 support so you could use ACE/TAO 7.0.8.

**Reply by cloudandzack007** [reply]
I can build the dds 3.21 with gcc 5.40，but the dds require glibc newer than 2.29。in my os(kylin4，chinese linux)，glibc is too old。If i try to update glibc，gcc will throw many error when build dds 3.21。

**Reply by mitza-oci** [reply]
We could probably help fix those errors, but we don't have sufficient details here to do so.

I don't recommend using a very old version of OpenDDS.  3.13 was release over 4 years ago.
